File: 704. 二分查找/main.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function (nums, target) {
  let left = 0,
    right = nums.length - 1;
  while (left <= right) {
    let mid = Math.floor((right - left) / 2) + left;
    const num = nums[mid];
    if (num == target) return mid;
    if (num < target) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }
  return -1;
};

var search2 = function (nums, target) {
  // 双重for循环
};

File: 704. 二分查找/main.rs
impl Solution {
    pub fn search(nums: Vec<i32>, target: i32) -> i32 {
        let left: i32 = 0;
        let right: i32 = nums.len() - 1;

        while left <= right {
            let mid = (right - left) / 2 + left;
            let num = nums[mid];
            if num == target {
                return mid;
            }
            if num > target {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        return -1;
    }
}
